core/c4diagram.go

```diff
@@ -0,0 +1,355 @@
+// package defines diagram as code using the graph input
+package core
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+	"unsafe"
+
+	"github.com/kislerdm/diagramastext/core/compression"
+)
+
+type optionsPlantUMLClient struct {
+	httpClient HttpClient
+}
+
+type clientPlantUML struct {
+	options optionsPlantUMLClient
+
+	baseURL string
+}
+
+func (c *clientPlantUML) Do(v DiagramGraph) (ResponseDiagram, error) {
+	code, err := diagramGraph2plantUMLCode(v)
+	if err != nil {
+		return nil, err
+	}
+	p, err := code2Path(code)
+	if err != nil {
+		return nil, err
+	}
+
+	diagram, err := c.requestHandler("svg/" + p)
+	if err != nil {
+		return nil, err
+	}
+
+	return ResponseC4Diagram{SVG: *(*string)(unsafe.Pointer(&diagram))}, nil
+}
+
+func (c *clientPlantUML) requestHandler(p string) ([]byte, error) {
```

Comment:
**question** `p` means path, param, or what?

Reply:
`p` means not much indeed :) Sorry, I'm used to "cut" variable names for private methods. What would you suggest? The context: route with the bite-encoded diagram code.

Reply:
I adjusted the attribute: it's called `encodedDiagramString`.